annif/cli.py

```diff
@@ -345,35 +364,51 @@ def run_learn(project_id, paths, docs_limit, backend_param):
 
 @cli.command("suggest")
 @click.argument("project_id")
+@click.argument("paths", type=click.Path(dir_okay=False, exists=True), nargs=-1)
 @click.option("--limit", "-l", default=10, help="Maximum number of subjects")
 @click.option("--threshold", "-t", default=0.0, help="Minimum score threshold")
 @click.option("--language", "-L", help="Language of subject labels")
+@click.option(
+    "--docs-limit",
+    "-d",
+    default=None,
+    type=click.IntRange(0, None),
+    help="Maximum number of documents to use",
+)
 @backend_param_option
 @common_options
-def run_suggest(project_id, limit, threshold, language, backend_param):
+def run_suggest(
+    project_id, paths, limit, threshold, language, backend_param, docs_limit
+):
     """
-    Suggest subjects for a single document from standard input.
+    Suggest subjects for a single document from standard input or for one or more
+    document file(s) given its/their path(s).
     \f
     This will read a text document from standard input and suggest subjects for
-    it.
+    it, or if given path(s) to file(s), suggest subjects for it/them.
     """
     project = get_project(project_id)
-    text = sys.stdin.read()
     lang = language or project.vocab_lang
     if lang not in project.vocab.languages:
         raise click.BadParameter(f'language "{lang}" not supported by vocabulary')
     backend_params = parse_backend_params(backend_param, project)
     hit_filter = SuggestionFilter(project.subjects, limit, threshold)
-    hits = hit_filter(project.suggest(text, backend_params))
-    for hit in hits.as_list():
-        subj = project.subjects[hit.subject_id]
-        click.echo(
-            "<{}>\t{}\t{}".format(
-                subj.uri,
-                "\t".join(filter(None, (subj.labels[lang], subj.notation))),
-                hit.score,
-            )
-        )
+
+    if click.get_text_stream("stdin").isatty():
```

Comment:
I don't think this is the correct test. Whether stdin is connected to a tty (e.g. run from a console session) should be irrelevant. Instead I think that `if paths:` or `if len(paths) > 0:` could work - meaning that if paths were given on the command line those should be used, but if not, fall back to reading stdin.